src/aes.rs

```diff
@@ -65,11 +65,11 @@ fn shift_rows(state: Block) -> Block {
     shift_row(3, 3, state)
 }
 
-fn xtime(x: u8) -> u8 {
+fn xtime(x: U8) -> U8 {
     let x1 = x << 1;
     let x7 = x >> 7;
-    let x71 = x7 & 1;
-    let x711b = x71 * 0x1b;
+    let x71 = x7 & U8(1);
```

Comment:
Can we do without the `U8`?

Reply:
We could if Rust allowed `&` operator overloading with heterogeneous types *à la Python* but unfortunately it does not: left and right operand have to be the same type, and since `x7` is a secret integr the `1` has to be made secret with `U8`. It is a reminder of the strong distinction that we impose between secret and public integers.

Reply:
But why does it work for `>>` then? We have `x71 = x7 & U8(1) = (x >> 7) & U8(1)`

Reply:
It works for `>>` because the right-side argument of `>>` is the number of bits to shift, and this number should be a public value between 0 and 32 for the whole operation to be constant-time. To ensure that it is a public value, we expect here the `u32` type instead of a secret type, so no casting needed.